lib/wasi/src/state/types.rs

```diff
@@ -218,13 +219,60 @@ pub(crate) fn poll(
     Ok(result.try_into().unwrap())
 }
 
-#[cfg(not(unix))]
+#[cfg(any(not(unix), not(feature = "sys-poll")))]
 pub(crate) fn poll(
-    _selfs: &[&dyn VirtualFile],
-    _events: &[PollEventSet],
-    _seen_events: &mut [PollEventSet],
-) -> Result<(), FsError> {
-    unimplemented!("VirtualFile::poll is not implemented for non-Unix-like targets yet");
+    files: &[&dyn VirtualFile],
+    events: &[PollEventSet],
+    seen_events: &mut [PollEventSet],
+) -> Result<u32, FsError> {
+    if !(files.len() == events.len() && events.len() == seen_events.len()) {
+        return Err(FsError::InvalidInput);
```

Comment:
It'd probably be a good idea to do a `debug!` print here to show the reason why it failed, since `InvalidInput` doesn't have a human readable description.